lib/seahorn/CexExeGenerator.cc

```diff
@@ -0,0 +1,568 @@
+#include "seahorn/CexExeGenerator.hh"
+
+namespace seahorn {
+namespace cexGen {
+namespace utils {
+Constant *exprToLlvm(Type *ty, Expr e, LLVMContext &ctx, const DataLayout &dl) {
+  if (isOpX<TRUE>(e)) {
+    // JN: getTypeStoreSizeInBits returns 8 for i1.
+    //     This causes later an error (only visible in debug mode)
+    //     when calling ConstantArray::get(AT, LLVMarray) because
+    //     the element type from AT will be i1 but the type of the
+    //     constant in LLVMarray will be i8 because of the use of
+    //     getTypeStoreSizeInBits.
+    //
+    // return Constant::getIntegerValue (ty,
+    // APInt(dl.getTypeStoreSizeInBits(ty), 1));
+    return ConstantInt::getTrue(ctx);
+  } else if (isOpX<FALSE>(e)) {
+    // return Constant::getNullValue (ty);
+    return ConstantInt::getFalse(ctx);
+  } else if (isOpX<MPZ>(e) || bv::is_bvnum(e)) {
+    expr::mpz_class mpz;
+    mpz = isOpX<MPZ>(e) ? getTerm<expr::mpz_class>(e)
+                        : getTerm<expr::mpz_class>(e->arg(0));
+    if (ty->isIntegerTy() || ty->isPointerTy()) {
+      // JN: I think we can have the same issue as above but for now I leave
+      // like it is.
+      return Constant::getIntegerValue(
+          ty, toAPInt(dl.getTypeStoreSizeInBits(ty), mpz));
+    }
+    llvm_unreachable("Unhandled type");
+  } else {
+    // HACK: in structures, first argument is usually the base value we are
+    // after
+    if (strct::isStructVal(e)) {
+      return exprToLlvm(ty, e->arg(0), ctx, dl);
```

Comment:
arg(0)

Reply:
good catch, will use mem manager to extract base address instead